### DISCUSSION TITLE
`DOMException` on changed value of bound `x-for` duplicates element

### DISCUSSION THREAD

**Discussion by fholmqvist** [alpinejs/alpine#2658] [7. Bugs (archived)]
**v2.3**

# Expected

Changing a number in an iterated model changes the number.

# Actual

The number is changed, followed by `Uncaught DOMException: An attempt was made to use an object that is not, or is no longer, usable`.

Changing again throws `Uncaught ReferenceError: example is not defined` and inserts a duplicated element above it. Original element stops working. Repeating the process in the new element continues in the same fashion.

Happens more frequently on keyboard input than inc/decrementing with mouse clicks.

# How to reproduce
```
<div x-data='{ examples: [{ number: 0 }] }'>
    <template x-for='example in examples'>
        <div x-bind='example'>
            <p x-text='example.number'></p>
            <input type='number' x-model.number='example.number'></input>
        </div>
    </template>
</div>
```
https://jsfiddle.net/us9dzcwp/

# Question

I am not sure if I'm using Alpine correctly here, I have a use case where I want to modify values of nested objects. Eg:
```
{ weeks: [ { sessions: [{ exercises: [{}] }] ]}
```
Where I need CRUD functionality on all levels of nesting.

Thank you for your time.

**Comment by SimoTod** [comment]
any reason why you are not using v3? v2.3.0 is quite old and a lot of bugs have been fixed since.
`x-bind='example'` doesn't make much sense, what are you trying to do with that call?

P.s.
not alpine related but input tags are self-closing so you don't need `</input>`

**Comment by fholmqvist** [comment]
@SimoTod Upgraded to v3 and the bug disappeared!

> x-bind='example' doesn't make much sense, what are you trying to do with that call?

If you don't bind it there, Alpine refuses to render any element after it. In the example above, the `input` tag never gets rendered without the div wrapping `x-bind='example'`, even in v3.

**Reply by SimoTod** [reply]
I'm not sure, x-bind is used to bind attributes and directives to the current element, it shouldn't affect a child tag. 
See  https://jsfiddle.net/vcmL02rd/

**Reply by fholmqvist** [reply]
@SimoTod Oh, I was removing the entire div. Is it so that Alpine operates on one element, thus needing to have several elements wrapped in one?

**Reply by SimoTod** [reply]
yes, x-for only supports a single child element so if you need more than one, they need to be wrapped in something

**Reply by fholmqvist** [reply]
Thank you Simon, appreciated :)
